### PR TITLE
fix(security): remove command test execution for 2.8.x - CVE 2019-16405

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

* Remove command test execution
* Remove dead code
* Apply sanitize on arguments
* Use camel case

**Fixes** CVE 2019-16405, thank to TheCyberGeek and Khaled

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x (master)

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 1]
```diff
@@ -37,55 +37,54 @@
...
 
-if (isset($_GET["command_id"])) {
-    $command_id = $_GET["command_id"];
-} elseif (isset($_POST["command_id"])) {
-    $command_id = $_POST["command_id"];
-} else {
-    $command_id = null;
-}
+$commandId = filter_var(
+    $_GET["command_id"] ?? $_POST["command_id"] ?? null,
```
Comment (sc979) [inline www/include/configuration/configObject/command/minHelpCommand.php]: null coalescent ternary isn't supported in php < 7

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 2]
```diff
@@ -37,55 +37,54 @@
...
 
-if (isset($_GET["command_name"])) {
-    $command_name = $_GET["command_name"];
-} elseif (isset($_POST["command_name"])) {
-    $command_name = $_POST["command_name"];
-} else {
-    $command_name = null;
-}
+$commandName = filter_var(
+    $_GET["command_name"] ?? $_POST["command_name"] ?? null,
```
Comment (sc979) [inline www/include/configuration/configObject/command/minHelpCommand.php]: null coalescent ternary isn't supported in php < 7

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 3]
```diff
@@ -37,55 +37,54 @@
...
      * Match if the first part of the path is a MACRO
      */
-    if (preg_match("/@DOLLAR@USER([0-9]+)@DOLLAR@/", $resource_def, $matches)) {
+    if (preg_match("/@DOLLAR@USER([0-9]+)@DOLLAR@/", $resourceDef, $matches)) {
         /*
          * Select Resource line
          */
-        $DBRESULT = $pearDB->query("SELECT `resource_line` FROM `cfg_resource` WHERE `resource_name` = '\$USER".$matches[1]."\$' LIMIT 1");
+        $res = $pearDB->query(
+            "SELECT `resource_line` FROM `cfg_resource` WHERE `resource_name` = '\$USER".$matches[1]."\$' LIMIT 1"
```
Comment (sc979) [inline www/include/configuration/configObject/command/minHelpCommand.php]:
```suggestion
            "SELECT `resource_line` FROM `cfg_resource` WHERE `resource_name` = '\$USER" . $matches[1] . "\$' LIMIT 1"
```

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 4]
```diff
@@ -37,55 +37,54 @@
...
-        unset($DBRESULT);
+        $resource = $res->fetchRow();
+        unset($res);
 
-        $resource_path = $resource["resource_line"];
-        unset($cmd_array[0]);
-        $command = rtrim($resource_path, "/")."#S#".implode("#S#", $cmd_array);
+        $resourcePath = $resource["resource_line"];
+        unset($aCmd[0]);
+        $command = rtrim($resourcePath, "/")."#S#".implode("#S#", $aCmd);
```
Comment (sc979) [inline www/include/configuration/configObject/command/minHelpCommand.php]:
```suggestion
        $command = rtrim($resourcePath, "/") . "#S#" . implode("#S#", $aCmd);
```

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 5]
```diff
@@ -38,54 +38,67 @@
...
     /*
      * Match if the first part of the path is a MACRO
      */
-    if (preg_match("/@DOLLAR@USER([0-9]+)@DOLLAR@/", $resource_def, $matches)) {
+    if (preg_match("/@DOLLAR@USER([0-9]+)@DOLLAR@/", $resourceDef, $matches)) {
         /*
          * Select Resource line
          */
-        $DBRESULT = $pearDB->query("SELECT `resource_line` FROM `cfg_resource` WHERE `resource_name` = '\$USER".$matches[1]."\$' LIMIT 1");
+        $res = $pearDB->query(
```
Comment (callapa) [inline www/include/configuration/configObject/command/minHelpCommand.php]: Use PDO::prepare instead

**Pull request merged**